app/routes/_gcn.circulars._archive._index/SortSelectorButton.tsx

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { useSubmit } from '@remix-run/react'
+import {
+  Button,
+  ButtonGroup,
+  CardBody,
+  Grid,
+  Icon,
+  Radio,
+} from '@trussworks/react-uswds'
+import classNames from 'classnames'
+import type { ChangeEvent } from 'react'
+import { useState } from 'react'
+
+import DetailsDropdownContent from '~/components/DetailsDropdownContent'
+
+const sortOptions = [
+  { id: 'radio-sort-circularId', value: 'circularID', label: 'Circular' },
+  { id: 'radio-sort-relevance', value: 'relevance', label: 'Relevance' },
+]
+
+function SortButton({
+  sort,
+  expanded,
+  ...props
+}: {
+  sort?: string
+  expanded?: boolean
+} & Omit<Parameters<typeof ButtonGroup>[0], 'segmented' | 'children'>) {
+  const slimClasses = 'height-4 padding-y-0'
+
+  return (
+    <ButtonGroup type="segmented" {...props}>
+      <Button type="button" className={`${slimClasses} padding-x-2`}>
+        Sorted By{' '}
+        {sortOptions.find((option) => option.value === sort)?.label ||
+          'Circular'}
+      </Button>
```

Comment:
Make this an object keyed by value so that you do not have to do a linear search to find the label:

```suggestion
const sortOptions = {
  circularID: 'Circular',
  relevance: 'Relevance',
}

function SortButton({
  sort,
  expanded,
  ...props
}: {
  sort?: string
  expanded?: boolean
} & Omit<Parameters<typeof ButtonGroup>[0], 'segmented' | 'children'>) {
  const slimClasses = 'height-4 padding-y-0'

  return (
    <ButtonGroup type="segmented" {...props}>
      <Button type="button" className={`${slimClasses} padding-x-2`}>
        Sorted By {sortOptions[sort ?? 'circular']}
      </Button>
```

Also I think it will be more clear if we call this Sort by Date rather than Sort by Circular.